fasthttpadaptor/request.go

```diff
@@ -0,0 +1,51 @@
+package fasthttpadaptor
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/valyala/fasthttp"
+)
+
+// ConvertRequest convert a fasthttp.Request to http.Request
+// setURI is optionally because your use in client mode
+func ConvertRequest(ctx *fasthttp.RequestCtx, setURI bool) (*http.Request, error) {
+	var r http.Request
```

Comment:
Maybe, the `http.Request` could be part of the `Convert Request()` parameters to can use a `sync.Pool` if it's needed.

Reply:
Somethig like this:

```go
func ConvertRequest(ctx *fasthttp.RequestCtx, setURI bool, req *http.Request) error
```

Reply:
I'm not sure I agree with this. `http.Request` contains unexported fields. It's not trivial to reuse an `http.Request` with `sync.Pool` and the standard library gives no guarantee that this might break in the future.

But I guess it's up to the user to make this work properly.